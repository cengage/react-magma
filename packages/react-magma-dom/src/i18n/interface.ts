import { Locale } from 'date-fns';

export interface I18nInterface {
  example: string;
  locale: Locale;
  dateFormat: 'MM/dd/yyyy' | 'dd/MM/yyyy' | 'yyyy/MM/dd' | 'yyyy/dd/MM';
  timeFormat: '12' | '24';
  months: {
    long: {
      january: string;
      february: string;
      march: string;
      april: string;
      may: string;
      june: string;
      july: string;
      august: string;
      september: string;
      october: string;
      november: string;
      december: string;
    };
    short: {
      january: string;
      february: string;
      march: string;
      april: string;
      may: string;
      june: string;
      july: string;
      august: string;
      september: string;
      october: string;
      november: string;
      december: string;
    };
  };
  days: {
    long: {
      monday: string;
      tuesday: string;
      wednesday: string;
      thursday: string;
      friday: string;
      saturday: string;
      sunday: string;
    };
    short: {
      monday: string;
      tuesday: string;
      wednesday: string;
      thursday: string;
      friday: string;
      saturday: string;
      sunday: string;
    };
    min: {
      monday: string;
      tuesday: string;
      wednesday: string;
      thursday: string;
      friday: string;
      saturday: string;
      sunday: string;
    };
  };
  emptyItemsListText: string;
  alert: {
    dismissAriaLabel: string;
  };
  breadcrumb: {
    navAriaLabel: string;
  };
  combobox: {
    clearIndicatorAriaLabel: string;
    createLabel: string;
  };
  datePicker: {
    startOfWeek:
      | 'sunday'
      | 'monday'
      | 'tuesday'
      | 'wednesday'
      | 'thursday'
      | 'friday'
      | 'saturday';
    calendarIconAriaLabel: string;
    calendarOpenAnnounce: string;
    calendarCloseAriaLabel: string;
    previousMonthAriaLabel: string;
    nextMonthAriaLabel: string;
    disabledDayAriaLabel: string;
    selectedDayAriaLabel: string;
    todayAriaLabel: string;

    helpModal: {
      header: string;
      helpButtonAriaLabel: string;
      enter: {
        ariaLabel: string;
        explanation: string;
      };
      rightAndLeftArrowKeys: {
        ariaLabel: string;
        explanation: string;
      };
      upAndDownArrowKeys: {
        ariaLabel: string;
        explanation: string;
      };
      pageUpAndPageDownKeys: {
        ariaLabel: string;
        displayValue: string;
        explanation: string;
      };
      homeAndEndKeys: {
        ariaLabel: string;
        displayValue: string;
        explanation: string;
      };
      escape: {
        ariaLabel: string;
        displayValue: string;
        explanation: string;
      };
      questionMark: {
        ariaLabel: string;
        explanation: string;
      };
    };
  };
  dropdown: {
    menuItemSelectedAriaLabel: string;
    toggleMenuAriaLabel: string;
  };
  header: {
    navigationButtonLabel: string;
    search: {
      input: {
        ariaLabel: string;
        placeholder: string;
      };
      iconAriaLabel: string;
    };
  };
  indeterminateCheckbox: {
    isCheckedAnnounce: string;
    isIndeterminateAnnounce: string;
    isUncheckedAnnounce: string;
  };
  loadingIndicator: {
    progressBar: {
      messages: {
        first: string;
        second: string;
        third: string;
      };
    };
    spinner: {
      messages: {
        first: string;
        second: string;
        third: string;
      };
    };
  };
  modal: {
    closeAriaLabel: string;
  };
  multiCombobox: {
    selectedItemButtonAriaLabel: string;
  };
  multiSelect: {
    placeholder: string;
    selectedItemButtonAriaLabel: string;
  };
  password: {
    shown: {
      ariaLabel: string;
      buttonText: string;
      announce: string;
    };
    hidden: {
      ariaLabel: string;
      buttonText: string;
      announce: string;
    };
  };
  search: {
    input: {
      ariaLabel: string;
      placeholder: string;
    };
    iconAriaLabel: string;
  };
  select: {
    placeholder: string;
    clearIndicatorAriaLabel: string;
  };
  skipLink: {
    buttonText: string;
  };
  spinner: {
    ariaLabel: string;
  };
  table: {
    pagination: {
      ofLabel: string;
      nextAriaLabel: string;
      previousAriaLabel: string;
      rowsPerPageLabel: string;
    };
  };
  tabs: {
    horizontalTabsInstructions: string;
    verticalTabsInstructions: string;
    previousButtonLabel: string;
    nextButtonLabel: string;
  };
  timePicker: {
    hoursAriaLabel: string;
    minutesAriaLabel: string;
    amButtonAriaLabel: string;
    pmButtonAriaLabel: string;
    amSelectedAnnounce: string;
    pmSelectedAnnounce: string;
  };
}
