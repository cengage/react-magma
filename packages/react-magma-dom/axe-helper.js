import { configureAxe } from 'jest-axe';

export const axe = configureAxe({
  rules: {
    // disabled landmark rules when testing isolated components.
    region: { enabled: false },
    // disabled nested-interactive when testing isolated components
    'nested-interactive': { enabled: false },
  },
});
