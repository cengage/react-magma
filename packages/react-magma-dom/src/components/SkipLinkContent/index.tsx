import * as React from 'react';

import { TARGET_ID } from '../SkipLink';

/**
 * @children required
 */
export interface SkipLinkContentProps
  extends React.HTMLAttributes<HTMLDivElement> {
  /**
   * @internal
   */
  testId?: string;
}

export const SkipLinkContent: React.FunctionComponent<
  SkipLinkContentProps
> = props => {
  const { children, testId } = props;
  return (
    <div
      data-testid={testId}
      id={TARGET_ID}
      tabIndex={-1}
      style={{ outline: 0 }}
    >
      {children}
    </div>
  );
};
