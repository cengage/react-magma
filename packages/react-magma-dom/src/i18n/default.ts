import { I18nInterface } from './interface';

export const defaultI18n: I18nInterface = {
  locale: null,
  dateFormat: 'MM/dd/yyyy',
  timeFormat: '12',
  example:
    'i18n defaults must be set in packages/react-magma-dom/src/i18n/default.ts',
  months: {
    long: {
      january: 'January',
      february: 'February',
      march: 'March',
      april: 'April',
      may: 'May',
      june: 'June',
      july: 'July',
      august: 'August',
      september: 'September',
      october: 'October',
      november: 'November',
      december: 'December',
    },
    short: {
      january: 'Jan',
      february: 'Feb',
      march: 'Mar',
      april: 'Apr',
      may: 'May',
      june: 'Jun',
      july: 'Jul',
      august: 'Aug',
      september: 'Sep',
      october: 'Oct',
      november: 'Nov',
      december: 'Dec',
    },
  },
  days: {
    long: {
      monday: 'Monday',
      tuesday: 'Tuesday',
      wednesday: 'Wednesday',
      thursday: 'Thursday',
      friday: 'Friday',
      saturday: 'Saturday',
      sunday: 'Sunday',
    },
    short: {
      monday: 'Mon',
      tuesday: 'Tues',
      wednesday: 'Wed',
      thursday: 'Thur',
      friday: 'Fri',
      saturday: 'Sat',
      sunday: 'Sun',
    },
    min: {
      monday: 'M',
      tuesday: 'T',
      wednesday: 'W',
      thursday: 'T',
      friday: 'F',
      saturday: 'S',
      sunday: 'S',
    },
  },
  emptyItemsListText: 'No Options',
  alert: {
    dismissAriaLabel: 'Close this message',
  },
  breadcrumb: {
    navAriaLabel: 'Breadcrumb',
  },
  charts: {
    line: {
      chartTabLabel: 'Chart',
      dataLegendsLabel:
        'Select one or more of the categories below to filter out the ones you don’t want to see.',
      dataTabLabel: 'Data',
      keyboardInstructions:
        'In the chart below, use your up and down arrows to move between each line. Use your left and right arrows to move between points on a line.',
      keyboardInstructionsHeader: 'Keyboard Instructions',
      keyboardInstructionsTooltip: 'Keyboard Instructions',
      legendButtonAriaLabel: 'Toggle data for {name}',
    },
  },
  combobox: {
    clearIndicatorAriaLabel:
      'reset selection for {labelText}. {selectedItem} is selected',
    createLabel: 'Create "{inputValue}"',
  },
  datePicker: {
    startOfWeek: 'sunday',
    calendarIconAriaLabel: 'Toggle Calendar Widget',
    calendarOpenAnnounce:
      'Calendar Widget is now open. Press the question mark key to get the keyboard shortcuts for changing dates.',
    calendarCloseAriaLabel: 'Close Calendar Widget',
    previousMonthAriaLabel: 'Previous Month',
    nextMonthAriaLabel: 'Next Month',
    disabledDayAriaLabel: 'Not Available. ',
    selectedDayAriaLabel: '(Selected)',
    todayAriaLabel: '(Today)',
    helpModal: {
      header: 'Keyboard Shortcuts',
      helpButtonAriaLabel: 'Calendar Widget Help',
      enter: {
        ariaLabel: 'Enter key',
        explanation: 'Select the date in focus.',
      },
      rightAndLeftArrowKeys: {
        ariaLabel: 'Right and left arrow keys',
        explanation: 'Move backward (left) and forward (right) by one day.',
      },
      upAndDownArrowKeys: {
        ariaLabel: 'up and down arrow keys',
        explanation: 'Move backward (up) and forward (down) by one week.',
      },
      pageUpAndPageDownKeys: {
        ariaLabel: 'page up and page down keys',
        displayValue: 'PGUP/PGDN',
        explanation: 'Switch months.',
      },
      homeAndEndKeys: {
        ariaLabel: 'Home and end keys',
        displayValue: 'HOME/END',
        explanation: 'Go to the first or last day of a week.',
      },
      escape: {
        ariaLabel: 'Escape key',
        displayValue: 'ESC',
        explanation: 'Return to the date input field.',
      },
      questionMark: {
        ariaLabel: 'Question Mark',
        explanation: 'Open this panel.',
      },
    },
  },
  dropdown: {
    menuItemSelectedAriaLabel: '(selected)',
    toggleMenuAriaLabel: 'Toggle menu',
  },
  dropzone: {
    browseFiles: 'Browse Files',
    dragMessage: 'Drag and drop or browse files to upload.',
    errors: {
      'too-many-errors': {
        message: 'Files must not have any errors.',
      },
      'too-many-files': {
        message: 'You must upload a maximum of',
      },
      'too-few-files': {
        message: 'You must upload a minimum of',
      },
      'file-invalid-type': {
        header: 'Invalid File Type',
        message: 'Upload only the allowed file types',
      },
      'file-too-large': {
        header: 'File exceeds size limit',
        message: 'Upload only files with a maximum size of',
      },
      'file-too-small': {
        header: 'File size is below the limit',
        message: 'Upload only files with a minimum size of',
      },
      required: {
        message: 'You must upload at least one file.',
      },
    },
    files: 'files',
    bytes: 'Bytes',
    deleteFile: 'Delete file',
    removeFile: 'Remove file',
  },
  header: {
    navigationButtonLabel: 'Open navigation menu',
    search: {
      input: {
        ariaLabel: 'Search',
        placeholder: 'Search',
      },
      iconAriaLabel: 'Search',
    },
  },
  indeterminateCheckbox: {
    isCheckedAnnounce: 'All subitems are checked for {labelText} checkbox',
    isIndeterminateAnnounce:
      'Some, but not all, subitems are checked for {labelText} checkbox',
    isUncheckedAnnounce: 'No subitems are checked for {labelText} checkbox',
  },
  input: {
    isClearableAriaLabel: 'Clear Input',
  },
  loadingIndicator: {
    progressBar: {
      messages: {
        first: 'Please be patient as this could take up to a minute to load.',
        second: 'Thank you for your patience. Still loading...',
        third: 'Thank you for waiting.  We’re almost there!',
      },
    },
    spinner: {
      messages: {
        first: 'Loading...',
        second: 'Thank you for your patience. Still loading...',
        third: 'Sorry for the delay. This is taking longer than expected.',
      },
    },
  },
  modal: {
    closeAriaLabel: 'Close dialog',
  },
  multiSelect: {
    placeholder: 'Select...',
    selectedItemButtonAriaLabel: 'reset item {selectedItem}',
  },
  multiCombobox: {
    selectedItemButtonAriaLabel: 'reset item {selectedItem}',
  },
  pagination: {
    nextButtonLabel: 'Next Page',
    previousButtonLabel: 'Previous Page',
    pageButtonLabel: 'Go to Page',
  },
  password: {
    shown: {
      ariaLabel:
        'Show password. Note: this will visually expose your password on the screen',
      buttonText: 'Show',
      announce: 'Password is now visible',
    },
    hidden: {
      ariaLabel: 'Hide password',
      buttonText: 'Hide',
      announce: 'Password is now hidden',
    },
  },
  search: {
    input: {
      ariaLabel: 'Search',
      placeholder: 'Search',
    },
    iconAriaLabel: 'Search',
  },
  select: {
    placeholder: 'Select...',
    clearIndicatorAriaLabel:
      'reset selection for {labelText}. {selectedItem} is selected',
  },
  skipLink: {
    buttonText: 'Skip Navigation',
  },
  spinner: {
    ariaLabel: 'Loading',
  },
  table: {
    pagination: {
      ofLabel: 'of',
      nextAriaLabel: 'Next page',
      previousAriaLabel: 'Previous page',
      rowsPerPageLabel: 'Rows per page',
    },
  },
  tabs: {
    horizontalTabsInstructions:
      'use the right and left arrow keys to activate other tabs',
    verticalTabsInstructions:
      'use the down and up arrow keys to activate other tabs',
    nextButtonLabel: 'Scroll tabs forward',
    previousButtonLabel: 'Scroll tabs back',
  },
  tag: {
    deleteAriaLabel: 'Delete {labelText} tag',
  },
  timePicker: {
    hoursAriaLabel: 'Hour',
    minutesAriaLabel: 'Minute',
    amButtonAriaLabel:
      'AM selected.  To change to PM press the p or enter button.',
    pmButtonAriaLabel:
      'PM selected.  To change to AM press the a or enter button.',
    amSelectedAnnounce: 'AM is now selected',
    pmSelectedAnnounce: 'PM is now selected',
  },
  wizard: {
    actions: {
      next: 'next',
      previous: 'previous',
      cancel: 'cancel',
      submit: 'submit',
    },
    optional: 'optional',
    navigationLabel: 'Navigation for the wizard',
  },
};
