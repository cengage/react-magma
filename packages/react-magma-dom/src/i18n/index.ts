import * as React from 'react';

import { defaultI18n } from './default';
import { I18nInterface } from './interface';

export const I18nContext = React.createContext<I18nInterface>(defaultI18n);
