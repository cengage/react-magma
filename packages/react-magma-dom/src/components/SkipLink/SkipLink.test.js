import React from 'react';
import { axe } from '../../../axe-helper';
import { SkipLink } from '.';
import { SkipLinkContent } from '../SkipLinkContent';

import { render, fireEvent } from '@testing-library/react';
import { I18nContext } from '../../i18n';
import { defaultI18n } from '../../i18n/default';

describe('SkipLink', () => {
  it('should find element by testId', () => {
    const testId = 'test-id';
    const { getByTestId } = render(<SkipLink testId={testId} />);

    expect(getByTestId(testId)).toBeInTheDocument();
  });

  it('should render the skip link component', () => {
    const { container } = render(<SkipLink />);
    const link = container.querySelector('a');

    expect(link).toBeInTheDocument();
    expect(link.innerHTML).toEqual('Skip Navigation');
    expect(link).toHaveStyleRule('background', '#3942B0');
    expect(link).toHaveStyleRule('color', '#FFFFFF');
    expect(link).toMatchSnapshot();
  });

  it('should render the skip link component with custom text', () => {
    const TEXT = 'Test text';
    const { container } = render(<SkipLink buttonText={TEXT} />);
    const link = container.querySelector('a');

    expect(link).toBeInTheDocument();
    expect(link.innerHTML).toEqual(TEXT);
  });

  it('should put focus on the h1 in the main content area when the skip link is clicked', () => {
    const { container } = render(
      <>
        <SkipLink />
        <SkipLinkContent>
          <h1>Test</h1>
        </SkipLinkContent>
      </>
    );
    const link = container.querySelector('a');
    const heading = container.querySelector('h1');

    expect(heading).not.toHaveFocus();
    fireEvent.click(link);
    expect(heading).toHaveFocus();
  });

  it('should put focus on the main content area when the skip link is clicked if they content area does not have an h1', () => {
    const { container } = render(
      <>
        <SkipLink />
        <SkipLinkContent />
      </>
    );
    const link = container.querySelector('a');
    const contentDiv = container.querySelector('#reactMagmaMainContent');

    expect(contentDiv).not.toHaveFocus();
    fireEvent.click(link);
    expect(contentDiv).toHaveFocus();
  });

  it('should not move focus if there is no content area', () => {
    const { container } = render(<SkipLink />);
    const link = container.querySelector('a');

    link.focus();
    fireEvent.click(link);
    expect(link).toHaveFocus();
  });

  it('should render the skip link with specified color and variant', () => {
    const { container } = render(
      <SkipLink color="success" variant="outline" />
    );
    const link = container.querySelector('a');

    expect(link).toHaveStyleRule('background', 'rgba(0,0,0,0)');
    expect(link).toHaveStyleRule('color', '#178037');
  });

  it('should render the skip link button the correct colors for an inverse button', () => {
    const { container } = render(<SkipLink isInverse />);
    const link = container.querySelector('a');

    expect(link).toHaveStyleRule('background', '#FFFFFF');
    expect(link).toHaveStyleRule('color', '#3942B0');
  });

  it('should render the skip link specified position top and left attributes', () => {
    const { container } = render(
      <SkipLink positionLeft={86} positionTop={99} />
    );
    const link = container.querySelector('a');

    expect(link).toHaveStyleRule('left', '86px', {
      target: ':focus',
    });
    expect(link).toHaveStyleRule('top', '99px', {
      target: ':focus',
    });
  });

  describe('i18n', () => {
    it('uses the context override for button text', () => {
      const buttonText = 'test button text';
      const { getByText } = render(
        <I18nContext.Provider
          value={{
            ...defaultI18n,
            skipLink: {
              buttonText,
            },
          }}
        >
          <SkipLink />
        </I18nContext.Provider>
      );

      expect(getByText(buttonText)).toBeInTheDocument();
    });
  });

  it('Does not violate accessibility standards', () => {
    const { container } = render(
      <>
        <SkipLink />
        <SkipLinkContent />
      </>
    );

    return axe(container.innerHTML).then(result => {
      return expect(result).toHaveNoViolations();
    });
  });
});
