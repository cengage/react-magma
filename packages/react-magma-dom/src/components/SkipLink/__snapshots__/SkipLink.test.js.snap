// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipLink should render the skip link component 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #3942B0;
  border: 0;
  border-color: #3942B0;
  border-radius: 4px;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-family: "Open Sans",Helvetica,sans-serif;
  font-size: 16px;
  font-weight: 600;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 24px;
  margin: 4px;
  min-width: 96px;
  overflow: hidden;
  padding: 12px 16px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  touch-action: manipulation;
  -webkit-transition: background 0.35s,border-color 0.35s,box-shadow 0.35s,color 0.35s;
  transition: background 0.35s,border-color 0.35s,box-shadow 0.35s,color 0.35s;
  vertical-align: middle;
  white-space: nowrap;
  width: auto;
  left: -9999px;
  position: fixed;
  top: -9999px;
}

.emotion-0:not(:disabled):focus {
  outline: 2px dotted #027EE1;
  outline-offset: 3px;
}

.emotion-0:not(:disabled):hover,
.emotion-0:not(:disabled):focus {
  background: #2d3489;
  color: #FFFFFF;
}

.emotion-0:not(:disabled):after {
  background: #FFFFFF;
  border-radius: 50%;
  content: '';
  height: 28px;
  left: 50%;
  opacity: 0;
  padding: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%) scale(1);
  -ms-transform: translate(-50%,-50%) scale(1);
  transform: translate(-50%,-50%) scale(1);
  -webkit-transition: opacity 1s,-webkit-transform 0.5s;
  -webkit-transition: opacity 1s,transform 0.5s;
  transition: opacity 1s,transform 0.5s;
  width: 28px;
}

.emotion-0:not(:disabled):active {
  background: #202563;
  color: #FFFFFF;
}

.emotion-0:not(:disabled):active:after {
  opacity: 0.4;
  -webkit-transform: translate(-50%,-50%) scale(0);
  -ms-transform: translate(-50%,-50%) scale(0);
  transform: translate(-50%,-50%) scale(0);
  -webkit-transition: -webkit-transform 0s;
  -webkit-transition: transform 0s;
  transition: transform 0s;
}

.emotion-0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-0:focus {
  left: 10px;
  top: 10px;
  z-index: 99999;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #3942B0;
  border: 0;
  border-color: #3942B0;
  border-radius: 4px;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-family: "Open Sans",Helvetica,sans-serif;
  font-size: 16px;
  font-weight: 600;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 24px;
  margin: 4px;
  min-width: 96px;
  overflow: hidden;
  padding: 12px 16px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  touch-action: manipulation;
  -webkit-transition: background 0.35s,border-color 0.35s,box-shadow 0.35s,color 0.35s;
  transition: background 0.35s,border-color 0.35s,box-shadow 0.35s,color 0.35s;
  vertical-align: middle;
  white-space: nowrap;
  width: auto;
  left: -9999px;
  position: fixed;
  top: -9999px;
}

.emotion-0:not(:disabled):focus {
  outline: 2px dotted #027EE1;
  outline-offset: 3px;
}

.emotion-0:not(:disabled):hover,
.emotion-0:not(:disabled):focus {
  background: #2d3489;
  color: #FFFFFF;
}

.emotion-0:not(:disabled):after {
  background: #FFFFFF;
  border-radius: 50%;
  content: '';
  height: 28px;
  left: 50%;
  opacity: 0;
  padding: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%) scale(1);
  -ms-transform: translate(-50%,-50%) scale(1);
  transform: translate(-50%,-50%) scale(1);
  -webkit-transition: opacity 1s,-webkit-transform 0.5s;
  -webkit-transition: opacity 1s,transform 0.5s;
  transition: opacity 1s,transform 0.5s;
  width: 28px;
}

.emotion-0:not(:disabled):active {
  background: #202563;
  color: #FFFFFF;
}

.emotion-0:not(:disabled):active:after {
  opacity: 0.4;
  -webkit-transform: translate(-50%,-50%) scale(0);
  -ms-transform: translate(-50%,-50%) scale(0);
  transform: translate(-50%,-50%) scale(0);
  -webkit-transition: -webkit-transform 0s;
  -webkit-transition: transform 0s;
  transition: transform 0s;
}

.emotion-0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-0:focus {
  left: 10px;
  top: 10px;
  z-index: 99999;
}

<a
  class="emotion-0 emotion-1"
  color="primary"
  href="#reactMagmaMainContent"
>
  Skip Navigation
</a>
`;
